calico_node/tests/st/bgp/test_ipip.py

```diff
@@ -402,21 +400,53 @@ def _test_gce_int(self, with_ipip, backend, host1, host2, rrc):
                                             ip_pass_list=[workload_host1.ip,
                                                           workload_host2.ip])
 
-                # Check that we are using IP-in-IP for some routes.
-                assert "tunl0" in host1.execute("ip r")
-                assert "tunl0" in host2.execute("ip r")
-
-                # Check that routes are not flapping: the following shell
-                # script checks that there is no output for 10s from 'ip
-                # monitor', on either host.  The "-le 1" is to allow for
-                # something (either 'timeout' or 'ip monitor', not sure) saying
-                # 'Terminated' when the 10s are up.  (Note that all commands
-                # here are Busybox variants; I tried 'grep -v' to eliminate the
-                # Terminated line, but for some reason it didn't work.)
                 for host in [host1, host2]:
-                    host.execute("changes=`timeout -t 10 ip -t monitor 2>&1`; " +
-                                 "echo \"$changes\"; " +
-                                 "test `echo \"$changes\" | wc -l` -le 1")
+
+                    # Check that we are using IP-in-IP for some routes.
+                    assert "tunl0" in host.execute("ip r")
+
+                    # Get current links and addresses, as a baseline for the
+                    # following changes test.
+                    host.execute("ip l")
+                    host.execute("ip a")
+
+                    # Check that routes are not flapping, on either host, by
+                    # running 'ip monitor' for 10s and checking that it does
+                    # not indicate any changes other than those associated with
+                    # an IP address being added to the tunl0 device.
+                    ip_changes = host.execute("timeout -t 10 ip -t monitor 2>&1 || true")
+                    lines = ip_changes.split("\n")
+                    assert len(lines) >= 1, "No output from ip monitor"
+                    expected_lines_following = 0
+                    for line in lines:
+                        expected = False
+                        if expected_lines_following > 0:
```

Comment:
The `expected_lines_following` could be avoided by just looping on an index variable and then any case that expects additional trailing lines could just advance the index the correct number of additional lines.  I think that might make the loop more readable, WDYT?  Though I guess looping with an index like that is less 'pythonic' than the current loop.

Reply:
I think I prefer to keep as is.  Please see what you think now that I've removed `expected` - I think it looks reasonably neat.